### PR TITLE
#3470 DNF_Sort with CPP

### PR DESCRIPTION
**Pull request by shubhgupta9**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This is Dutch National Flag sorting algorithm code in C++.
#3470 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3470

### CODE REVIEW

File: C-Plus-Plus/sort/DNF_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Created by shubhgupta9
+//C++ implementation of DNF sort or Dutch National Flag sort
+
+//code for the algorithm goes here:-
+
+/* This algorithm is used to sort an array of 3 numbers, in our example code we will be taking 
+a default unsorted array which consists of 0s,1s and 2s.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/DNF_sort.cpp]: - Add Multi-Line Comments

File: C-Plus-Plus/sort/DNF_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#include<iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/DNF_sort.cpp]: Remove `#include<iostream>` as you already use `#include<bits/stdc++.h>`

File: C-Plus-Plus/sort/DNF_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    arr[a] = arr[b];
+    arr[b] = temp;
+}
+
+//sorting algo 
+void DNF(int arr[],int n)
+{
+    int low = 0; // will signify from where the 1s will be starting 
+    int mid = 0; // denotes the unknown region 
+    int high = n-1; // will signify from where the 2s will be starting
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/DNF_sort.cpp]: Remove the redundant comments

File: C-Plus-Plus/sort/DNF_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+/*
+example input:-
+1 0 2 1 0 1 2 1 2 
+
+example output:-
+0 0 1 1 1 1 2 2 2
+
+time complexity: O(n)
+space compexity: O(1)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/DNF_sort.cpp]: Capitalize the first letter

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Kindly update the readme.md. Don't update the main readme file. If you find difficulty how to update and which readme to update Kindly see the any merged PR.

**Comment by shubhgupta9** [comment]
Sure sir.

File: C-Plus-Plus/sort/DNF_sort.cpp (hunk 1 above)
Comment (shubhgupta9) [reply C-Plus-Plus/sort/DNF_sort.cpp]: Thank you sir for reviewing my pull request.
I have done all the required changes.
I am not able to edit the README file please help me regarding that.
Thank You

Comment (shubhgupta9) [reply C-Plus-Plus/sort/DNF_sort.cpp]: Done Sir. You can see all the required changes.
Thank You for the review.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code already exists [here](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/other/SortColors-DutchNationalFlag.cpp)

**Pull request closed without merging**